File: Lab1/Lab1/Problem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

[assembly: InternalsVisibleTo("TestProject1")]
[assembly: InternalsVisibleTo("TestProject"), InternalsVisibleTo("GUI")]

namespace Lab1
{
    internal class Problem
    {
        int n;
        Random random;
        public List<Item> items = new List<Item>();

        public Problem(int n, int seed)
        {
            this.n = n;
            random = new Random(seed);
            MakeList();
        }

        void MakeList()
        {
            for (int i = 0; i < n; i++)
            {
                Item item = new Item(random.Next(1,9), random.Next(1,9), i);
                items.Add(item);
            }
        }

        public void ShowList()
        {
            for (int i = 0; i < n; i++)
            {
                Console.Write("item: ");
                Console.Write(items[i].id);
                Console.Write(", weight: ");
                Console.Write(items[i].weight);
                Console.Write(" value: ");
                Console.Write(items[i].value);
                Console.Write(" ratio: ");
                Console.WriteLine(items[i].ratio);
            }
        }


        public Result Solve(int capacity)
        {
            items.Sort((a, b) => a.ratio.CompareTo(b.ratio));
            items.Sort((a, b) => b.ratio.CompareTo(a.ratio));

            Result result = new Result(capacity);

            int amount = 0;
            bool success = true;

            while (amount < capacity && success)
            {
                success = false;
                for (int i = 0; i < items.Count; i++)
                {
                    if (items[i].used == false && items[i].weight <= capacity - amount)
                    {
                        items[i].used = true;
                        amount += items[i].weight;
                        result.backpack.Add(items[i]);
                    }
                }
            }
            result.Measure();
            return result;

        }
    }
}

File: Lab5/src/main/java/org/example/Problem.java
package org.example;

import java.util.*;

public class Problem
{
    public int n;
    public int capacity;
    List<Item> items = new ArrayList<>();
    Random random;

    Problem(int n, int cap, int seed, int lowerBound, int upperBound)
    {
        this.n = n;
        capacity = cap;
        random = new Random(seed);
        MakeProblem(lowerBound, upperBound);
    }

    private void MakeProblem(int lowerBound, int upperBound)
    {
        for (int i = 0; i < n; i++)
        {
            Item item = new Item(random.nextInt(lowerBound,upperBound), random.nextInt(lowerBound,upperBound), i);
            items.add(item);
        }
    }

    public void Solve()
    {
        //items.sort((s1, s2) -> s1.ratio - s2.ratio);
        Collections.sort(items, new Comparator<Item>()
        {
            @Override
            public int compare(Item o1, Item o2) {
                return Float.compare(o2.ratio, o1.ratio);
            }
        });

    }

    @Override
    public String toString()
    {
        String s = "";
        for (int i = 0; i < n; i++)
        {
            s += "item: ";
            s += items.get(i).id;
            s += ",\t weight: ";
            s += items.get(i).weight;
            s += "\t value: ";
            s += items.get(i).value;
            s += "\t ratio: ";
            s += items.get(i).ratio+"\n";
        }
        return s;
    }
}
